pkg/pgmodel/migrate.go

```diff
@@ -313,47 +366,90 @@ func replaceSchemaNames(r io.ReadCloser) (string, error) {
 	return s, err
 }
 
+//A migration file is inside a directory that is a semver version number. The filename itself has the format
+//<migration file number)-<description>.sql. That file correspond to the semver of <dirname>.<migration file number>
+//where the migration file number is always part of prerelease tag.
+//All app versions >= (inclusive) migration files's semver will include the migration file
+func (t *Migrator) getMigrationFileVersion(dirName string, fileName string) (*semver.Version, error) {
+	var migrationFileNumber int
+	matches := migrationFileNameRegexp.FindStringSubmatch(fileName)
+	if len(matches) < 2 {
+		return nil, fmt.Errorf("unable to parse the migration file name %v", fileName)
+	}
+	n, err := fmt.Sscanf(matches[1], "%d", &migrationFileNumber)
+	if n != 1 || err != nil {
+		return nil, fmt.Errorf("unable to parse the migration file name %v: %w", fileName, err)
+	}
+
+	migrationFileVersion, err := semver.Make(dirName)
+	if err != nil {
+		return nil, fmt.Errorf("unable to parse version from directory %v: %w", dirName, err)
+	}
+	migrationNumberPrVersion, err := semver.NewPRVersion(fmt.Sprintf("%d", migrationFileNumber))
```

Comment:
I'm inclined to read `PrVersion` as "dull request version" instead of "pre-release version", maybe `DevVersion` instead?